### PR TITLE
Adding native dependency injection table

### PR DESCRIPTION
**Pull request by krmahadevan**

### CODE REVIEW

**Comment by krmahadevan** [comment]
@cbeust - Can you please help review and accept this PR ?

**Comment by juherr** [comment]
About the annotation order, I prefer the following: https://github.com/cbeust/testng/wiki/Allowed-parameters-injection-in-methods

**Comment by krmahadevan** [comment]
@juherr - I have re-ordered the elements in the table.

File: doc/documentation-main.html [hunk 1]
```diff
@@ -2218,6 +2218,118 @@ <h3 class="sourcetitle">Shell</h3>
...
+			<th>&nbsp;ITestContext&nbsp;</th>
+			<th>&nbsp;XmlTest&nbsp;</th>
+			<th>&nbsp;Method&nbsp;</th>
+			<th>&nbsp;Object[]&nbsp;</th>
+			<th>&nbsp;ITestResult&nbsp;</th>
+		</tr>
+		<tr>
+			<td>BeforeSuite</td>
+			<td>Yes</td>
+			<td>Yes</td>
```
Comment (cbeust) [inline doc/documentation-main.html]: Let's say "No" here, per our discussion.

File: doc/documentation-main.html [hunk 2]
```diff
@@ -2218,6 +2218,118 @@ <h3 class="sourcetitle">Shell</h3>
...
+			<td>Test</td>
+			<td>Yes</td>
+			<td>No</td>
+			<td>No</td>
+			<td>No</td>
+			<td>No</td>
+		</tr>
+		<tr>
+			<td>DataProvider</td>
+			<td>Yes</td>
```
Comment (cbeust) [inline doc/documentation-main.html]: Same, "No" here.

File: doc/documentation-main.html (hunk 1 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: Fixed.

File: doc/documentation-main.html (hunk 2 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: @cbeust - Why here ? I didn't get you.

**Comment by krmahadevan** [comment]
@cbeust - I have made the relevant changes.

**Pull request merged**